baremetal/machine_manager_test.go

```diff
@@ -25,11 +41,32 @@ const (
 	testUserDataSecretName = "worker-user-data"
 )
 
+var CloudInitData = []byte("metal3:cloudInitData1010101test__hello")
+
 var ProviderID = "12345ID6789"
+var newProviderID = "changedID12345"
+
 var bmmSpec = &capbm.BareMetalMachineSpec{
 	ProviderID: &ProviderID,
 }
 
+var bmmSpecNil = &capbm.BareMetalMachineSpec{
+	ProviderID: nil,
+}
+
+var bmmSpecAll = &capbm.BareMetalMachineSpec{
+	ProviderID: &ProviderID,
+	UserData: &corev1.SecretReference{
+		Name:      "mybmmachine",
+		Namespace: "myns",
+	},
+	Image: capbm.Image{
+		URL:      testImageURL,
+		Checksum: testImageChecksumURL,
+	},
+	HostSelector: capbm.HostSelector{},
+}
+
```

Comment:
I am wondering if declaring these different bmmspecs would make more sense if it is done in testcases.

Reply:
The structs defined here are mainly used in two or more test cases, avoiding repetition.
I would prefer to have it like this.